solitude/exceptions.py

```diff
@@ -4,8 +4,17 @@
 from lib.bango.errors import BangoImmediateError
 from lib.brains.errors import BraintreeResultError
 
+from django.db.transaction import rollback
+
+from solitude.logger import getLogger
+
+log = getLogger('s')
+
 
 def custom_exception_handler(exc):
+    log.warning('Transaction rolled back due to error.')
+    rollback()
```

Comment:
I don't know if this will do what you want (because I'm not sure if transactions are enabled like this for mysql). Also, it doesn't seem to be testable probably because custom error handlers aren't triggered from the test client, I think. What if we turned on per request transactions? https://docs.djangoproject.com/en/1.8/topics/db/transactions/#tying-transactions-to-http-requests